### ISSUE TITLE
(suggestion) adding beforeSend to Ajaxes requests

### ISSUE THREAD

**Issue by mdarabi** [smartystreets/jquery.liveaddress#50]
This is a just a suggestion to overwrite any existing headers for $.ajax. By adding `beforeSend(xhr) {}` to Ajaxes requests, you can make sure nothing unnecessary (that might cause CORS issue) will be sent to the server.

**Comment by camiblanch** [comment]
I'm not sure exactly how this would be implemented. Could you help me understand with some specifics? Are you suggesting that possibly on line 970 and line 2370 we add something like `beforeSend(xhr){}`?

**Comment by mdarabi** [comment]
As you mentioned, if you just add beforeSend(xhr){} to $.ajax in lines 970 and 2370, it would overwrite any predefined beforeSend functions (for example, those from ajaxSetup) and will remove any additional headers. 
In my case, because we have one global ajaxSetup (we know, this is not a good practice but we had to do it), live-address plugin could not send requests and so it did not work. We had to host the plugin and add beforeSend(xhr){} to both ajax calls to solve the problem.
